mono/metadata/class-internals.h

```diff
@@ -609,22 +602,23 @@ struct _MonoGenericContainer {
 	   the generic container of the containing class. */
 	MonoGenericContainer *parent;
 	/* the generic type definition or the generic method definition corresponding to this container */
+	/* Union rules: If is_anonymous, image field is valid; else if is_method, method field is valid; else klass is valid. */
 	union {
 		MonoClass *klass;
 		MonoMethod *method;
+		MonoImage *image;
 	} owner;
-	int type_argc    : 31;
+	int type_argc    : 29; // Per the ECMA spec, this value is capped at 16 bits
 	/* If true, we're a generic method, otherwise a generic type definition. */
 	/* Invariant: parent != NULL => is_method */
-	int is_method    : 1;
+	int is_method     : 1;
+	/* If true, this is one of the special anonymous containers in MonoImage, or a container created by SRE. */
+	int is_anonymous : 1;
+	/* If false, all params in this container are full-size. If true, all params are just param structs. */
+	/* This field is always == to the is_anonymous field except during container creation. */
```

Comment:
"except during container creation" Why? For how long? How will code witness this transitory state?

Please expand this comment on what's the issue around container creation.
Doing it while the knowledge is fresh will save us from a lot of trouble down the line.